timesketch/lib/aggregators/bucket.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Bucket aggregations."""
+
+from __future__ import unicode_literals
+
+from timesketch.lib.aggregators import manager
+from timesketch.lib.aggregators import interface
+
+
+class TermsAggregation(interface.BaseAggregator):
+    """Terms Bucket Aggregation."""
+
+    NAME = 'bucket_terms'
+    SUPPORTED_CHARTS = frozenset(['barchart', 'horizontal_barchart'])
+    FORM_FIELDS = {
+        'field': {
+            'type': 'text',
+            'description': 'What field to aggregate.'
+        },
+        'limit': {
+            'type': 'number',
+            'description': 'Number of results to return.'
+        }
+    }
+
+    def __init__(self, sketch_id=None, index=None):
+        """Initialize the aggregation object.
+
+        Args:
+            sketch_id: Sketch ID as string.
+            index: List of indexes to run the aggregation on.
+        """
+        super(TermsAggregation, self).__init__(sketch_id, index)
```

Comment:
you are not doing anything in the __init__ here, you can just skip the`_init__` s

Reply:
I'm initialising my super class with sketch_id and index, so this is needed.

Reply:
well... no... it will get passed to super you are not making any changes, if you skip it, it will inherit the __init__ from super, which is the same as you are doing, you are just adding an __init__ just to call super, this is necessary if you are making changes ... as in creating a new attribute, etc, etc

Reply:
As discussed offline. This is an old habit of mine. I have removed the redundant __init__